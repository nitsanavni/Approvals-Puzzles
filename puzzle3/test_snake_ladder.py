from snake_ladder import start
from approvaltests import verify
import random


def test_start(mocker):
    mocker.patch("snake_ladder.get_player_names",
                 return_value=("player_a", "player b"))
    mocker.patch("time.sleep")
    mocker.patch("builtins.input")
    spy = mocker.patch("builtins.print")
    # mocker.patch("sys.exit")
    random.seed(2)

    try:
        start()
    except:
        pass

    verify(spy.call_args_list)
