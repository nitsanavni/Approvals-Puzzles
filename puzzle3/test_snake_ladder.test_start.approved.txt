[call('\n    Welcome to Snake and Ladder Game.\n    Version: 1.0.0\n    Developed by: https://www.pythoncircle.com\n    \n    Rules:\n      1. Initally both the players are at starting position i.e. 0. \n         Take it in turns to roll the dice. \n         Move forward the number of spaces shown on the dice.\n      2. If you lands at the bottom of a ladder, you can move up to the top of the ladder.\n      3. If you lands on the head of a snake, you must slide down to the bottom of the snake.\n      4. The first player to get to the FINAL position is the winner.\n      5. Hit enter to roll the dice.\n    \n    '),
 call('\nRolling dice...'),
 call('Its a 1'),
 call('player_a moving....'),
 call('\nplayer_a moved from 0 to 1'),
 call('\nRolling dice...'),
 call('Its a 3'),
 call('player b moving....'),
 call('\nplayer b moved from 0 to 3'),
 call('\nWOWW ########'),
 call('\nplayer b climbed the ladder from 3 to 20'),
 call('\nRolling dice...'),
 call('Its a 6'),
 call('player_a moving....'),
 call('\nplayer_a moved from 1 to 7'),
 call('\nRolling dice...'),
 call('Its a 3'),
 call('player b moving....'),
 call('\nplayer b moved from 20 to 23'),
 call('\nRolling dice...'),
 call('Its a 2'),
 call('player_a moving....'),
 call('\nplayer_a moved from 7 to 9'),
 call('\nRolling dice...'),
 call('Its a 1'),
 call('player b moving....'),
 call('\nplayer b moved from 23 to 24'),
 call('\nRolling dice...'),
 call('Its a 6'),
 call('player_a moving....'),
 call('\nplayer_a moved from 9 to 15'),
 call('\nWOWW ########'),
 call('\nplayer_a climbed the ladder from 15 to 34'),
 call('\nRolling dice...'),
 call('Its a 6'),
 call('player b moving....'),
 call('\nplayer b moved from 24 to 30'),
 call('\nRolling dice...'),
 call('Its a 6'),
 call('player_a moving....'),
 call('\nplayer_a moved from 34 to 40'),
 call('\nRolling dice...'),
 call('Its a 3'),
 call('player b moving....'),
 call('\nplayer b moved from 30 to 33'),
 call('\nRolling dice...'),
 call('Its a 4'),
 call('player_a moving....'),
 call('\nplayer_a moved from 40 to 44'),
 call('\nRolling dice...'),
 call('Its a 3'),
 call('player b moving....'),
 call('\nplayer b moved from 33 to 36'),
 call('\nRolling dice...'),
 call('Its a 1'),
 call('player_a moving....'),
 call('\nplayer_a moved from 44 to 45'),
 call('\nRolling dice...'),
 call('Its a 4'),
 call('player b moving....'),
 call('\nplayer b moved from 36 to 40'),
 call('\nRolling dice...'),
 call('Its a 4'),
 call('player_a moving....'),
 call('\nplayer_a moved from 45 to 49'),
 call('\nOH MY GOD... ########'),
 call('\nplayer_a climbed the ladder from 49 to 67'),
 call('\nRolling dice...'),
 call('Its a 2'),
 call('player b moving....'),
 call('\nplayer b moved from 40 to 42'),
 call('\nRolling dice...'),
 call('Its a 2'),
 call('player_a moving....'),
 call('\nplayer_a moved from 67 to 69'),
 call('\nRolling dice...'),
 call('Its a 2'),
 call('player b moving....'),
 call('\nplayer b moved from 42 to 44'),
 call('\nRolling dice...'),
 call('Its a 2'),
 call('player_a moving....'),
 call('\nplayer_a moved from 69 to 71'),
 call('\nRolling dice...'),
 call('Its a 2'),
 call('player b moving....'),
 call('\nplayer b moved from 44 to 46'),
 call('\nRolling dice...'),
 call('Its a 5'),
 call('player_a moving....'),
 call('\nplayer_a moved from 71 to 76'),
 call('\nRolling dice...'),
 call('Its a 3'),
 call('player b moving....'),
 call('\nplayer b moved from 46 to 49'),
 call('\nYAAYYY ########'),
 call('\nplayer b climbed the ladder from 49 to 67'),
 call('\nRolling dice...'),
 call('Its a 5'),
 call('player_a moving....'),
 call('\nplayer_a moved from 76 to 81'),
 call('\nWOWW ########'),
 call('\nplayer_a climbed the ladder from 81 to 98'),
 call('\nRolling dice...'),
 call('Its a 4'),
 call('player b moving....'),
 call('\nplayer b moved from 67 to 71'),
 call('\nRolling dice...'),
 call('Its a 5'),
 call('player_a moving....'),
 call('You need 2 to win this game. Keep trying.'),
 call('\nRolling dice...'),
 call('Its a 5'),
 call('player b moving....'),
 call('\nplayer b moved from 71 to 76'),
 call('\nRolling dice...'),
 call('Its a 3'),
 call('player_a moving....'),
 call('You need 2 to win this game. Keep trying.'),
 call('\nRolling dice...'),
 call('Its a 2'),
 call('player b moving....'),
 call('\nplayer b moved from 76 to 78'),
 call('\nRolling dice...'),
 call('Its a 6'),
 call('player_a moving....'),
 call('You need 2 to win this game. Keep trying.'),
 call('\nRolling dice...'),
 call('Its a 4'),
 call('player b moving....'),
 call('\nplayer b moved from 78 to 82'),
 call('\nRolling dice...'),
 call('Its a 5'),
 call('player_a moving....'),
 call('You need 2 to win this game. Keep trying.'),
 call('\nRolling dice...'),
 call('Its a 4'),
 call('player b moving....'),
 call('\nplayer b moved from 82 to 86'),
 call('\nRolling dice...'),
 call('Its a 4'),
 call('player_a moving....'),
 call('You need 2 to win this game. Keep trying.'),
 call('\nRolling dice...'),
 call('Its a 5'),
 call('player b moving....'),
 call('\nplayer b moved from 86 to 91'),
 call('\nRolling dice...'),
 call('Its a 6'),
 call('player_a moving....'),
 call('You need 2 to win this game. Keep trying.'),
 call('\nRolling dice...'),
 call('Its a 4'),
 call('player b moving....'),
 call('\nplayer b moved from 91 to 95'),
 call('\nRolling dice...'),
 call('Its a 5'),
 call('player_a moving....'),
 call('You need 2 to win this game. Keep trying.'),
 call('\nRolling dice...'),
 call('Its a 5'),
 call('player b moving....'),
 call('\nplayer b moved from 95 to 100'),
 call('\n\n\nThats it.\n\nplayer b won the game.'),
 call('Congratulations player b'),
 call('\nThank you for playing the game. Please visit https://www.pythoncircle.com\n\n')]
